scipy/interpolate/_bsplines.py

```diff
@@ -561,6 +561,87 @@ def integrate(self, a, b, extrapolate=None):
         integral *= sign
         return integral.reshape(ca.shape[1:])
 
+    def design_matrix(self, x, kind="dense", check_finite=True):
+        '''
+        Returns a design matrix for BSpline object in "dense" "CSR" formats.
+        
+        Parameters
+        ----------
+        x : 1-D array, shape(n,)
+            Values of x - coordinate of a given set of points (n >= 1).
```

Comment:
Please check that `x` is indeed 1D and raise if it's not. Otherwise the error message is not very user-friendly

```
In [43]: b.design_matrix([[1.5, 2], [1, 1]], kind='CSR')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-43-b84b44aae402> in <module>
----> 1 b.design_matrix([[1.5, 2], [1, 1]], kind='CSR')

~/repos/scipy/scipy/build/testenv/lib/python3.8/site-packages/scipy/interpolate/_bsplines.py in design_matrix(self, x, kind, check_finite)
    627             col_ind = []
    628             for i in range(n):
--> 629                 ind = find_left(t, x[i], k)
    630                 vals = _bspl.evaluate_all_bspl(t, k, x[i], ind)
    631                 # special check due to periodic boundary conditions:

~/repos/scipy/scipy/build/testenv/lib/python3.8/site-packages/scipy/interpolate/_bsplines.py in find_left(t, x, k)
    598             nt = len(t)
    599             for i in range(nt - 1):
--> 600                 if (t[i] <= x and t[i + 1] > x):
    601                     return i
    602             return nt - k - 2

ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```